docs/NETWORK-ACLS.md

```diff
@@ -0,0 +1,289 @@
+
+Every ACE should have a unique non-zero integer ID assigned. Please note that ACE ID should not exceed 24bits.
+ACE ID is set in [flowlogs](../api/proto/flowlog/flowlog.proto) as `aclId` for every flow to show which ACE was applied.
+ACE `ID=0` is reserved for a default reject-all rule that implicitly exists at the end of every ACL (even flows whose
+packets are rejected are logged).
+
+## ACL Semantics
+
+Every ECO interface is assigned a single separate access control list filtering inbound and outbound traffic
+(unless it is a passthrough interface, in which case EVE networking is completely bypassed).
+A single entry of ACL, denoted as ACE, consists of a set of matches that use layer 3 and 4 information to select
+a subset of the traffic, and a set of actions deciding what to do with the selected traffic. Optionally, it is possible
+to limit ACE to only ingress or egress flow direction. By default, ACEs are bi-directional, i.e. applied irrespective
+of the flow direction.
+
+For every packet leaving or entering ECO, the list of ACEs assigned to the corresponding interface is traversed
+sequentially in the order as configured. The fate of a packet is determined by the first ACE in the list for which
+the L3 & L4 packet header fields satisfy every match entry (i.e., matches are ANDed and empty set matches any packet).
+Packet can be allowed to continue (`ALLOW` action), dropped (`DROP` action) or mangled - specifically
+the destination IP address and port can be rewritten to allow an inbound flow to traverse the NAT of a local network
+(`PORTMAP` action).
+
+Apart from stateless filtering, EVE also supports ACL-based traffic limiting based
+on the [Token bucket algorithm](https://en.wikipedia.org/wiki/Token_bucket). ACE with the `LIMIT` action
+will either accept or reject packet depending on the current state of token buckets to keep the packet rate
+of the connection within the configured limits.
+
+Packet that does not match any configured ACE is rejected. In other words, an implicit ACE with an empty set of matches
+and the `DROP` action is present at the very end of every ACL. This means that by default application will not be
+able to connect to any other deployed application or external endpoint. The only exceptions are local services
+provided by EVE, such as DNS server, DHCP server or the internal HTTP server providing cloud-init metadata,
+all of which are always accessible to the application (and cannot be blocked).
+
+ACE which is configured without specifying the action will `ALLOW` matched packets. The reasoning is that with
+the implicit reject-all ACE at the end of every ACL, it is expected that majority of explicitly configured rules
+will be used to list the set of endpoints that application is *allowed to* communicate with.
+
+To summarize, ACL is composed of user-configured ACEs and some implicit rules as follows:
+
+```text
+ACL:
+  - implicit ACEs to permit DNS, DHCP, internal HTTP, IPv6 ND
```

Comment:
Could spell out ND as Neighbor Discovery to keep yetus happy.

Reply:
done :)